usaspending_api/transactions/delta_models/transaction_search.py

```diff
@@ -1,104 +1,491 @@
+    "cfda_title": "STRING",
+    "cfda_id": "INTEGER",
+    "pop_country_name": "STRING",
+    "pop_country_code": "STRING",
+    "pop_state_name": "STRING",
+    "pop_state_code": "STRING",
+    "pop_county_code": "STRING",
+    "pop_county_name": "STRING",
+    "pop_zip5": "STRING",
+    "pop_congressional_code": "STRING",
+    "pop_congressional_population": "INTEGER",
+    "pop_county_population": "INTEGER",
+    "pop_state_fips": "STRING",
+    "pop_state_population": "INTEGER",
+    "pop_city_name": "STRING",
+    "recipient_location_country_code": "STRING",
+    "recipient_location_country_name": "STRING",
+    "recipient_location_state_name": "STRING",
+    "recipient_location_state_code": "STRING",
+    "recipient_location_state_fips": "STRING",
+    "recipient_location_state_population": "INTEGER",
+    "recipient_location_county_code": "STRING",
+    "recipient_location_county_name": "STRING",
+    "recipient_location_county_population": "INTEGER",
+    "recipient_location_congressional_code": "STRING",
+    "recipient_location_congressional_population": "INTEGER",
+    "recipient_location_zip5": "STRING",
+    "recipient_location_city_name": "STRING",
+    "recipient_hash": "STRING",
+    "recipient_levels": "ARRAY<STRING>",
+    "recipient_name": "STRING",
+    "recipient_unique_id": "STRING",
+    "parent_recipient_hash": "STRING",
+    "parent_recipient_name": "STRING",
+    "parent_recipient_unique_id": "STRING",
+    "recipient_uei": "STRING",
+    "parent_uei": "STRING",
+    "awarding_toptier_agency_id": "INTEGER",
+    "funding_toptier_agency_id": "INTEGER",
+    "awarding_agency_id": "INTEGER",
+    "funding_agency_id": "INTEGER",
+    "awarding_toptier_agency_name": "STRING",
+    "funding_toptier_agency_name": "STRING",
+    "awarding_subtier_agency_name": "STRING",
+    "funding_subtier_agency_name": "STRING",
+    "awarding_toptier_agency_abbreviation": "STRING",
+    "funding_toptier_agency_abbreviation": "STRING",
+    "awarding_subtier_agency_abbreviation": "STRING",
+    "funding_subtier_agency_abbreviation": "STRING",
+    "treasury_account_identifiers": "ARRAY<INTEGER>",
+    "tas_paths": "ARRAY<STRING>",
+    "tas_components": "ARRAY<STRING>",
+    "federal_accounts": "STRING",
+    "disaster_emergency_fund_codes": "ARRAY<STRING>",
+    "awarding_office_code": "STRING",
+    "awarding_office_name": "STRING",
+    "funding_office_code": "STRING",
+    "funding_office_name": "STRING",
+}
+
+transaction_search_create_sql_string = fr"""
+    CREATE OR REPLACE TABLE {{DESTINATION_TABLE}} (
+        {", ".join([f'{key} {val}' for key, val in _transaction_search_types.items()])}
     )
     USING DELTA
-    LOCATION 's3a://{SPARK_S3_BUCKET}/{DELTA_LAKE_S3_PATH}/{DESTINATION_DATABASE}/{DESTINATION_TABLE}'
-    """
+    LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
+"""
+
+transaction_search_load_sql_string = fr"""
+    INSERT OVERWRITE rpt.transaction_search
```

Comment:
So does the `OVERWRITE` keyword here effectively do the same as `saveMode("overwrite")` in the pyspark syntax? ... 
And ... does that mean it marks all prior parquet files for that table as inactive (to be vacuumed) and everything that is part of _this_ INSERT becomes the _only_ data in the table?

Reply:
Ok, i think those docs make it clear to me. But we ought to be careful about writing with partitions. Maybe it just works out though, that we'd only want to truncate data for changed partitions. But perhaps a row's partition column changed ; we'd need to make sure everything was wiped, not just the new partition (a lot like the ES requirement to delete-before-adding-in-order-to-update ... due to data landing on different shards).

So all that to mean, we may want to see if there's a `DELETE FROM` or `TRUNCATE` command we _explicitly_ run ahead of the `INSERT` on the _WHOLE_ table if we truly want to blow away the whole table.

Reply:
Talked about it and it seems that there's no _implied_ partition (i.e. looking up the partition(s) affected and just replacing those based on the partition_col value of the rows inserted) ... you have to _explicitly_ tell it what partition you want to affect, if you want to affect just 1 partition. 

Still worth a test to confirm that assumption about the docs but code should be safe as is.